src/workerd/jsg/rtti.h

```diff
@@ -578,7 +578,11 @@ struct MemberCounter {
   template <typename Type, typename GetNamedMethod, GetNamedMethod getNamedMethod>
   inline void registerWildcardProperty() { /* not a member */ }
 
-  template <const char* name, typename Method, Method method>
+  template <const char* name,
+      typename Method,
+      Method method,
+      bool failOnIncompatibleFastApi = false,
```

Comment:
Why do we need this? This adds a lot of noise to the code.

Reply:
JSG_FAST_XXX methods should fail if the function signature is not compatible with fast api. This was a request by @danlapid to understand/debug/enforce a method to validate if it's fast api compatible or not.

Reply:
Do we think people are going to want to _assert_ that certain methods are fast API? I kind of feel like nobody will care to do that. But if we do think that, then maybe we can have a separate macro like `JSG_ASSERT_FASTAPI(method)` which performs a static_assert. This can work independently of the whole registration system.

Reply:
Yeah I like the idea of using JSG_ASSERT_FASTAPI. I moved the implementation to there.